src/buildstream_plugins/sources/cargo.py

```diff
@@ -75,6 +85,10 @@
 from buildstream import Source, SourceFetcher, SourceError
 from buildstream import utils
 
+import dulwich
```

Comment:
For buildstream-plugins, we prefer to have no dependencies outside of the standard library.

Some exceptions have been made for tomli and requests, which we expect to be API stable, any unstable APIs we rely on can cause BuildStream projects to be dependent on host specific configurations, and even worse, when junctions are involved this can result in being painted into a corner where multiple projects require different versions of the same external python libraries, and thus cannot be built.

This has caused much issues specifically with `dulwich` being used by the `git_repo` plugin in bst-plugins-experimental, which appears to be a wildly changing API and the specific version of dulwich to use is not very clear.

Reply:
The bigger problem to be honest is lack of tests with the plugins involved. Even if minimum version is set (like it is now), nothing proves it actually works.

Reply:
> For buildstream-plugins, we prefer to have no dependencies outside of the standard library.

Maybe we can just move this as a new plugin to the experimental plugins repo? The git plugin that is actually used is already there and uses dulwich as well.

Reply:
The experimental repo is indeed the wild west, no objections to that.

The issue is that even with python libraries which conform well to pythonic standards of API compatibility, and follow semantic versioning, are not strict enough to be safe to use in a BuildStream plugin strictly speaking, since it is impossible to address both API versions separately (import names have not changed) or install both versions in a single python environment.

This will be equally true of course for the `bst-plugins-experimental` plugins, but we are not concerned with long term compatibility there.